### PR TITLE
OHRM5X-2106: Hash oauth - client secret

### PR DESCRIPTION
**Pull request by RajithaKumara**

### CODE REVIEW

File: src/client/src/orangehrmCoreOAuthPlugin/pages/oauth/OAuthAuthorize.vue [hunk 1]
```diff
@@ -19,27 +19,32 @@
...
-      <p>{{ index }}. {{ name }}: {{ value }}</p>
-    </div>
-    <oxd-button
-      display-type="ghost"
-      :label="$t('general.cancel')"
-      @click="onCancel"
-    />
-    <submit-button :label="$t('general.submit')" />
-  </oxd-form>
+  <oxd-text v-if="errorType !== null" tag="h6">
```
Comment (RajithaKumara) [inline src/client/src/orangehrmCoreOAuthPlugin/pages/oauth/OAuthAuthorize.vue:22]: this screen is dummy screen for the consent screen, @devishke-orange will complete it

File: src/client/src/orangehrmCoreOAuthPlugin/pages/registerOAuthClient/OAuthClient.vue [hunk 2]
```diff
@@ -186,7 +186,7 @@ export default {
     onClickDeleteSelected() {
       const ids = [];
       this.checkedItems.forEach((index) => {
-        ids.push(this.items?.data[index].clientId);
+        ids.push(this.items?.data[index].id);
```
Comment (RajithaKumara) [inline src/client/src/orangehrmCoreOAuthPlugin/pages/registerOAuthClient/OAuthClient.vue:189]: Delete functionality not worked

File: src/client/src/orangehrmCoreOAuthPlugin/pages/registerOAuthClient/SaveOAuthClient.vue [hunk 3]
```diff
@@ -22,12 +22,16 @@
...
+          editMode
+            ? $t('admin.edit_oauth_client')
+            : $t('admin.add_oauth_client')
+        }}
       </oxd-text>
 
       <oxd-divider />
 
-      <oxd-form :loading="isLoading" @submit-valid="onSave">
+      <oxd-form novalidate="true" :loading="isLoading" @submit-valid="onSave">
```
Comment (RajithaKumara) [inline src/client/src/orangehrmCoreOAuthPlugin/pages/registerOAuthClient/SaveOAuthClient.vue:34]: @Super-Chama when do we need `novalidate="true"`, it came from OAuth edit screen

File: src/plugins/orangehrmCoreOAuthPlugin/test/Api/OAuthClientAPITest.php [hunk 4]
```diff
@@ -57,7 +58,18 @@ public function testUpdate(TestCaseParams $testCaseParams): void
...
+        $api = $this->getApiEndpointMockBuilder(
+            OAuthClientAPI::class,
+            [
+                RequestParams::PARAM_TYPE_ATTRIBUTE => $testCaseParams->getAttributes() ?? [],
+                RequestParams::PARAM_TYPE_QUERY => $testCaseParams->getQuery() ?? [],
+                RequestParams::PARAM_TYPE_BODY => $testCaseParams->getBody() ?? [],
+            ]
+        )->onlyMethods(['generateSecret'])
+            ->getMock();
+        $api->expects($this->atMost(1))
```
Comment (RajithaKumara) [inline src/plugins/orangehrmCoreOAuthPlugin/test/Api/OAuthClientAPITest.php:70]: @devishke-orange seems not added tests for `create` operation. If you had a issue with dynamic client id creation, mocking is the solution, check this

File: src/client/src/orangehrmCoreOAuthPlugin/pages/registerOAuthClient/SaveOAuthClient.vue (hunk 3 above)
Comment (Super-Chama) [reply src/client/src/orangehrmCoreOAuthPlugin/pages/registerOAuthClient/SaveOAuthClient.vue:34]: no need it's already added in oxd form https://github.com/orangehrm/oxd/blob/main/components/src/core/components/Form/Form.vue#L25

File: src/plugins/orangehrmCoreOAuthPlugin/test/Api/OAuthClientAPITest.php (hunk 4 above)
Comment (devishke-orange) [reply src/plugins/orangehrmCoreOAuthPlugin/test/Api/OAuthClientAPITest.php:70]: ah thanks. I did try mocking but could not get it to work at that time. will try again

**Pull request merged**